### PR TITLE
Add GlobalState in OfflinePlugin

### PR DESCRIPTION
**Pull request by amitkma**
https://github.com/GetStream/stream-chat-android/issues/2889

### 🎯 Goal

Create GlobalState in OfflinePlugin where all data sources from the ChatDomain interface will be stored. ChatDomain should be wrapper with this state usage under the hood

### 🛠 Implementation details

- Added a `GlobalState` class that uses ChatDomain to provide state values.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalState.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import io.getstream.chat.android.client.models.ChannelMute
+import io.getstream.chat.android.client.models.Mute
+import io.getstream.chat.android.client.models.TypingEvent
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.offline.model.ConnectionState
+import io.getstream.chat.android.offline.utils.Event
+import kotlinx.coroutines.flow.StateFlow
+
+/**
+ * Global state of the Chat domain.
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalState.kt:13]: Maybe we can rephrase it to:
`[OfflinePlugin] global state.`

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalState.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ */
+public interface GlobalState {
+
+    /** The current user on the chatDomain object */
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalState.kt:17]: Let's add missing @return parameters and puncations in all those comments

Comment (amitkma) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalState.kt:17]: These are variables and not functions. Should we add `@return` to them?

**Review by adasiewiczr — COMMENTED** [review]
Let's move `GlobalState` to `StateRegistry` and we're good to go

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by adasiewiczr — APPROVED** [review]
